robot/server/server.go

```diff
@@ -322,20 +325,19 @@ func (s *Server) StreamStatus(req *pb.StreamStatusRequest, streamServer pb.Robot
 	ticker := time.NewTicker(every)
 	defer ticker.Stop()
 	for {
-		select {
-		case <-streamServer.Context().Done():
+		if !utils.SelectContextOrWaitChan(streamServer.Context(), ticker.C) {
 			return streamServer.Context().Err()
-		default:
-		}
-		select {
-		case <-streamServer.Context().Done():
-			return streamServer.Context().Err()
-		case <-ticker.C:
```

Comment:
Refactored to use the `SelectContextOrWaitChan` function which does the exact the same thing.